public/app/features/plugins/tests/datasource_srv.test.ts

```diff
@@ -251,6 +270,16 @@ describe('datasource_srv', () => {
         expect(exprWithName?.uid).toBe(ExpressionDatasourceRef.uid);
         expect(exprWithName?.type).toBe(ExpressionDatasourceRef.type);
       });
+
+      it('should return settings for runtime datasource when called with uid', () => {
+        const settings = dataSourceSrv.getInstanceSettings(runtimeDataSource.uid);
+        expect(settings).toBe(runtimeDataSource.instanceSettings);
+      });
+
+      it('should return settings for runtime datasource when called with name', () => {
```

Comment:
@torkelo I do understand why you didn't add support for this since it isn't a unique identifier in its current form. 

Question is, isn't it a risk that we try to fetch settings via name somewhere (older code) and in that scenario won't get a match. Or is this something that we don't have to worry about? If that is the case I will revert this test.

Reply:
no don't think that is a worry, everything should go through DataSourceSrv.

I am more worried that multiple plugins are going to register a data source with the same name, but maybe unlikely :)

Reply:
Ok ok, then I will skip this test. Any other tests you can think of that we should add in this PR?